src/games/strategy/triplea/ai/proAI/ProAI.java

```diff
@@ -226,8 +197,8 @@ protected void purchase(final boolean purchaseForBid, int PUsToSpend, final IPur
       } finally {
         data.releaseReadLock();
       }
-      this.data = dataCopy;
-      s_battleCalculator.setGameData(dataCopy);
+      ProData.setData(dataCopy);
```

Comment:
See if you can avoid this. The way `cloneGameData` data works is by serialized the gameData object to disk and then reading it back. For a larger game file this can take seconds, a performance killer.